src/compiler/crystal/macros.cr

```diff
@@ -1232,8 +1239,12 @@ module Crystal::Macros
   # class ExceptionHandler < ASTNode
   # end
 
-  # class ProcLiteral < ASTNode
-  # end
+  class ProcLiteral < ASTNode
+    # Return the function declaration for this proc.
+    # The name of the function will be "->"
+    def method : Def
```

Comment:
This is actually the way a ProcLiteral is implemented. This shouldn't leak to the macros API. A ProcLiteral should have the usual methods a `def` has. The macro logic just needs to forward some methods to the internal `def`.

What I mean is, `ProcLiteral` should have a `body`, `args`, etc. methods. It shouldn't have a `name`, though, because proc literals don't have a name.

Reply:
Is the way I have done that now ok, or should I do every method explicitly?

(Is there some consequence of the way that I've done it that isn't desirable?)

Reply:
I'm not sure. There are many methods in the base ASTNode class. The safest thing to do is to only forward the names that `Def` also handles.

Reply:
I've altered that to pass through all the current methods on `Def`, otherwise call `super`.